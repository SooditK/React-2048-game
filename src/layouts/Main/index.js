import React from 'react';
import { Switch, Route } from 'react-router-dom';
import App from '../../containers/App';
import Comments from '../../components/Comments';
import Ranking from '../../containers/Ranking';

const Main = () => (
  <main>
    <Switch>
      <Route exact path="/" component={App} />
      <Route exact path="/comments" component={Comments} />
      <Route exact path="/ranking" component={Ranking} />
      {/* 404 Page */}
      <Route component={App} />
    </Switch>
  </main>
);

export default Main;
