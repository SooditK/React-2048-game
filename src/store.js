import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import undoable from 'redux-undo';
import rootReducer from './reducers';
import rootSaga from './sagas';

const initHistory = JSON.parse(localStorage.getItem('state') || 'null');

const sagaMiddleware = createSagaMiddleware();
/* eslint-disable no-underscore-dangle */
const args = [
  undoable(rootReducer, {
    limit: 11, // set a limit for the history
    ignoreInitialState: true
  })
];
// Load localStorage data if available
if (initHistory) {
  args.push(initHistory);
}

if (window.__REDUX_DEVTOOLS_EXTENSION__) {
  args.push(
    compose(
      applyMiddleware(sagaMiddleware),
      // Redux devtools necessary code
      window.__REDUX_DEVTOOLS_EXTENSION__()
    )
  );
} else {
  args.push(applyMiddleware(sagaMiddleware));
}

const store = createStore(...args);
sagaMiddleware.run(rootSaga);

// Call this function while redux state changed,
// this callback save redux state to localStorage
store.subscribe(() => {
  const state = store.getState();
  localStorage.setItem('state', JSON.stringify(state));
});

export default store;
