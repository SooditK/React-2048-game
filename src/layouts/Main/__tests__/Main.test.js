import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import Main from '..';
import store from '../../../store';

// Make random always return 0.5, so snapshot always same.
const mockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath;

describe('<Main />', () => {
  it('component render', () => {
    const main = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Main />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(main).toMatchSnapshot();
  });
});
