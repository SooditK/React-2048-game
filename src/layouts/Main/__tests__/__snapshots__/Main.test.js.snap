// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> component render 1`] = `
<main>
  <div>
    <div
      className="app"
    >
      <div
        className="box"
      >
        <div
          className="board"
        >
          <h1
            className="title"
          >
            2048
          </h1>
          <table
            className="board"
          >
            <tbody>
              <tr>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell color-4"
                  >
                    <div
                      className="number"
                    >
                      4
                    </div>
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  <div
                    className="cell color-4"
                  >
                    <div
                      className="number"
                    >
                      4
                    </div>
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
                <td>
                  <div
                    className="cell"
                  >
                    <div
                      className="number"
                    />
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          className="panel"
        >
          <div>
            <div
              className="index"
            >
              <p
                className="score"
              >
                SCORE
                <em
                  className="text"
                >
                  0
                </em>
              </p>
              <p
                className="score best"
              >
                BEST
                <em
                  className="text"
                >
                  0
                </em>
              </p>
            </div>
          </div>
          <div
            className="panel"
          >
            <div
              className="rowBtn"
            >
              <div
                className="btn"
              >
                <button
                  className="btn-flat btn-md btn-default"
                  onClick={[Function]}
                >
                  <img
                    alt="speaker"
                    src="speaker-on.svg"
                  />
                </button>
              </div>
              <div
                className="btn "
              >
                <button
                  className="btn-flat btn-md btn-default"
                  onClick={[Function]}
                >
                  <img
                    alt="undo"
                    src="undo.svg"
                  />
                </button>
              </div>
              <div
                className="btn "
              >
                <button
                  className="btn-flat btn-md btn-default"
                  onClick={[Function]}
                >
                  <img
                    alt="reset"
                    src="reset.svg"
                  />
                </button>
              </div>
            </div>
            <div
              className="btn "
            >
              <button
                className="btn-flat btn-md btn-primary"
                onClick={[Function]}
              >
                <img
                  alt="arrow up"
                  className="up"
                  src="arrow.svg"
                />
              </button>
            </div>
            <div
              className="arrows"
            >
              <div
                className="btn "
              >
                <button
                  className="btn-flat btn-md btn-primary"
                  onClick={[Function]}
                >
                  <img
                    alt="arrow left"
                    className="left"
                    src="arrow.svg"
                  />
                </button>
              </div>
              <div
                className="btn "
              >
                <button
                  className="btn-flat btn-md btn-primary"
                  onClick={[Function]}
                >
                  <img
                    alt="arrow down"
                    className="down"
                    src="arrow.svg"
                  />
                </button>
              </div>
              <div
                className="btn "
              >
                <button
                  className="btn-flat btn-md btn-primary"
                  onClick={[Function]}
                >
                  <img
                    alt="arrow right"
                    className="right"
                    src="arrow.svg"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="tips"
      >
        <p
          className="title"
        >
          Tips
        </p>
        <p
          className="content"
        >
          Use keyboard arrow keys (or \`WASD\` if you like) control blocks;Click undo button to revert to last step status if you regretsã€‚You score will upload to ranking list after you login at \`Comments\` panel
        </p>
      </div>
      <div>
        <div
          className="hidden"
        >
          <div
            className="overlay"
          />
          <div
            className="modal center"
          >
            <div>
              <div
                className="gameover"
              >
                <div
                  className="text"
                >
                   
                  SCORE
                   
                </div>
                <div
                  className="banner"
                >
                  <p>
                    0
                  </p>
                </div>
              </div>
              <div
                className="buttonWrapper"
              >
                <button
                  className="button"
                  onClick={[Function]}
                >
                  <img
                    alt="reset"
                    src="reset.svg"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
